compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -2198,10 +2271,10 @@ impl std::fmt::Debug for Type {
             Type::Error => write!(f, "error"),
             Type::NamedGeneric(binding, name, kind) => match kind {
                 Kind::Normal => {
-                    write!(f, "{} -> {:?}", name, binding)
+                    write!(f, "{}{:?}", name, binding)
                 }
                 Kind::Numeric(typ) => {
-                    write!(f, "({} : {}) -> {:?}", name, typ, binding)
```

Comment:
This formatting was incorrect previously since the `->` is saying the named generic was bound to that type variable rather than just saying it is that type variable.